File: warmup/time-conversion.clj
(defn timeConversion [s]
    (def ampm (apply str (take-last 2 s)))
    (def hours (Integer/parseInt (apply str (take 2 s))))
    (def time-part (drop-last 2 (drop 2 s)))
    (defn make-time [hours] (concat (format "%02d" hours) time-part))
    (apply str (cond 
        (and (= hours 12) (= ampm "AM")) (make-time 0)
        (and (= hours 12) (= ampm "PM")) (make-time 12)
        (= ampm "PM") (make-time (+ 12 hours))
        :else (make-time hours)))
)

; Standard boilerplate
(def fptr (get (System/getenv) "OUTPUT_PATH"))

(def s (read-line))

(def result (timeConversion s))

(spit fptr (str result "\n") :append true)

File: warmup/time-conversion.js
'use strict';


function dropAmPm(time) {
    return time.substr(0, time.length - 2);
}

function timeConversion(s) {
    const amPM = s.substr(-2);

    const isTwelveOClock = s.substr(0, 2) === '12';

    const isMidnight = isTwelveOClock && amPM === 'AM';
    if (isMidnight) {
        return '00' + s.substr(2, s.length - 4);
    }

    const isNoon = isTwelveOClock && amPM === 'PM';
    if (isNoon || amPM === 'AM') {
        return dropAmPm(s);
    }
    
    const hours = (Number(s.substr(0, 2)) + 12).toString().padStart(2, '0');

    return hours + s.substr(2, s.length - 4);
}

// Standard boilerplate

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(str => str.trim());

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const s = readLine();

    let result = timeConversion(s);

    ws.write(result + "\n");

    ws.end();
}
